### ISSUE TITLE
Floating Windows Stutter/Lag when Moved/Resized

### ISSUE THREAD

**Issue by simboid** [qtile/qtile#1584]
When I move/resize windows they lag/stutter really badly; sometimes if you really quickly either resize or move a floating window the whole wm will freeze for a couple seconds. This is especially prevalent with heavier programs for some reason. Also; somewhat unrelated; when resizing if the resized by the corner closest to your mouse instead of bottom right all the time it would feel a lot more natural.

**Comment by tych0** [comment]
On Sun, Apr 05, 2020 at 11:05:22AM -0700, simboid wrote:
> When I move/resize windows they lag/stutter really badly; sometimes if you really quickly either resize or move a floating window the whole wm will freeze for a couple seconds. This is especially prevalent with heavier programs for some reason. Also; somewhat unrelated; when resizing if the resized by the corner closest to your mouse instead of bottom right all the time it would feel a lot more natural.

Probably because we do an actual resize, vs. just drawing a box
showing where it would be resized to if you released right now.

If someone wanted to send a patch for this, that would be great :)

**Comment by ramnes** [comment]
That being said IIRC other WM also do the actual resize and have much less lag. Could we optimize without losing the content, for example by drawing the window only once every ms or something in those lines?

**Comment by pawamoy** [comment]
I like what @ramnes is proposing, sometimes you resize based on how the program is rearranging its components (typically: a responsive page in a browser).

**Comment by m-col** [comment]
Implementing X sync extension protocol might help, e.g.: https://www.youtube.com/watch?v=js4UsY2m0wo

**Comment by nonetrix** [comment]
> According to [a Reddit post](https://reddit.com/r/qtile/comments/shfdzr/buggy_movementresizing_any_ideas/hvbf0nz/), and from my understanding, very high mouse polling rate appears to be the cause of the issue. It'd be great if resizing rate could be adjusted.

Had this exact issue in BSPWM no fix in it either 

I can trigger it by running Alacritty and running Cava at 165 FPS which stresses the terminal

**Comment by StarTroop** [comment]
I suppose this wouldn't be trivial, but would a potential solution be to change it so that moving or resizing windows only draws the border until the window is set? Personally, I'd consider the constant redrawing of moving windows a form of eye-candy, so I think it better fits the scope of a lightweight tiling WM to be as responsive as possible under any circumstances.

**Comment by m-col** [comment]
Interesting idea - but I think (for x11) this would require work by the compositor rather than the WM. The borders are essentially attached to the window, rather than drawn manually by us.

**Comment by StarTroop** [comment]
Hmm, I just tested moving a window on my laptop (where I don't use picom), and there is no lag on windows I know to lag on my desktop (with picom). I wonder if there's a setting for picom that can fix the problem.

**Comment by m-col** [comment]
It could also be the polling rate of your laptop's trackpad - if it was less frequent than your desktop's mouse then there would be less frequency updates for the same movement distance

**Comment by StarTroop** [comment]
Ah, yes I believe you're right, my desktop mouse is set to 1000hz, and I just found the other issue on that topic. I will say though, that the lag was noticeably improved when I disabled picom on my desktop, though not eliminated. I haven't found a way to achieve what I asked with picom yet, so I'd like to hear if anyone has ideas for workarounds for this issue, even if its "ugly".

**Comment by charbelnicolas** [comment]
I don't know how many times I have installed qtile to instantly uninstall it because of this issue. Thought it would be fixed by now, I guess there's no reason to move from awesome wm.

I use NVIDIA proprietary drivers on Arch Linux with a logitech mouse with 1000Hz polling rate with picom vsync enabled. So I guess I checked all the right boxes. This only happens on Qtile! Not on leftwm, awesome wm, i3, gnome, etc...
